clients/naming_client/push_receiver.go

```diff
@@ -103,6 +103,16 @@ func (us *PushReceiver) startServer() {
 }
 
 func (us *PushReceiver) handleClient(conn *net.UDPConn) {
+
+	if conn == nil {
+		ok := false
+		conn, ok = us.tryListen()
```

Comment:
tryListen should be extract from StartServer() func. because us.tryListen () does not change the port.In Handleclient, you also need to interval the fetch of conn, for example, 5S once.Otherwise, the outer for loop will consume more resources.

应该将startServer func中获取tryListen 提取func, 目前这样直接tryListen 并没有换端口操作。 在handleClient中还需要将获取conn的操作进行间隔，例如5S一次。否则外层for循环会消耗较多资源。

Reply:
Okay yes, I got your point. One thing what do you mean when you say  "you also need to interval the fetch of conn, for example, 5S once" .